src/redux/mainWallet/selectors/balances.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2017–2018, LaborX PTY
+ * Licensed under the AGPL Version 3 license.
+ */
+
+import { createSelector } from 'reselect'
+import { getTokens } from 'redux/tokens/selectors'
+import { selectMainWalletBalancesListStore, selectMarketPricesListStore, selectMarketPricesSelectedCurrencyStore } from 'redux/wallet/selectors'
+import { tokensAndAmountsSelector, tokensCountBalanceSelector } from './tokens'
+
+export const filteredBalances = (blockchain: string, filterSymbol) => createSelector(
+  [
+    selectMainWalletBalancesListStore,
+    getTokens,
+  ],
+  (
+    balances,
+    tokens,
+  ) => {
+    return balances
+      .filter((balance) => {
+        const symbol = balance.symbol()
+        const token = symbol && tokens.item(symbol)
+        return token && token.blockchain() === blockchain && (filterSymbol ? symbol === filterSymbol : true)
+      })
+  },
+)
+
+export const filteredBalancesAndTokens = (blockchain: string, symbol: string) => createSelector(
+  [
+    filteredBalances(blockchain, symbol),
+    getTokens,
+  ],
+  (
+    balances,
+    tokens,
+  ) => {
+    return balances
+      .map((balance) => {
+        return {
+          balance: balance,
+          token: tokens.item(balance.symbol()),
+        }
+      })
+  },
+)
+
+export const balanceCalculator = (blockchain: string, symbol: string) => createSelector(
```

Comment:
Are you really using balanceCalculator outside? Maybe it is better to export only necessary selectors?